src/transformers/generation/configuration_utils.py

```diff
@@ -0,0 +1,545 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Generation configuration class and utilities."""
+
+import copy
+import json
+import os
+from typing import Any, Dict, Optional, Union
+
+from .. import __version__
+from ..utils import (
+    GENERATION_CONFIG_NAME,
+    PushToHubMixin,
+    cached_file,
+    download_url,
+    extract_commit_hash,
+    is_remote_url,
+    logging,
+)
+
+
+logger = logging.get_logger(__name__)
+
+
+class GenerationConfig(PushToHubMixin):
+    r"""
+    Class that holds a configuration for a generation task.
+
+    <Tip>
+
+    A configuration file can be loaded and saved to disk. Loading and using a generation configuration file does
+    **not** change a model configuration or weights. It only affects the model's behavior at generation time.
+
+    </Tip>
+
+    Arg:
+        max_length (`int`, *optional*, defaults to `20`):
```

Comment:
```suggestion
        max_length (`int`, *optional*, defaults to 20):
```
No code formatting for numbers (here and all instances below)